Missing httpx dependency in tests

The default API client tests fail due to a missing `httpx` dependency.

The error from gitlab CI is:

```
ImportError while importing test module '/builds/**********************/tests/test_api.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_api.py:1: in <module>
    from fastapi.testclient import TestClient
/usr/local/lib/python3.11/site-packages/fastapi/testclient.py:1: in <module>
    from starlette.testclient import TestClient as TestClient  # noqa
/usr/local/lib/python3.11/site-packages/starlette/testclient.py:16: in <module>
    import httpx
E   ModuleNotFoundError: No module named 'httpx'
```

This is a documented dependency of FastAPI's test client: https://fastapi.tiangolo.com/tutorial/testing/. It needs to be added to the test dependencies in `pyproject.toml`.

--- a/{{ cookiecutter.repo_name }}/pyproject.toml
+++ b/{{ cookiecutter.repo_name }}/pyproject.toml
@@ -28,6 +28,7 @@ dependencies = [
 # Set any development related dependencies here
 [project.optional-dependencies]
 tests = [
+    "httpx",
     "pytest",
     "pytest-cov",
     "requests",

